sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/ConfidentialLedgerFailoverService.cs

```diff
@@ -0,0 +1,262 @@
+                                    {
+                                        failoverLedgerId = nameProp.GetString();
+                                    }
+                                    else
+                                    {
+                                        foreach (JsonProperty prop in failoverLedger.EnumerateObject())
+                                        {
+                                            if (prop.Value.ValueKind == JsonValueKind.String && string.Equals(prop.Name, "id", StringComparison.OrdinalIgnoreCase))
+                                            {
+                                                failoverLedgerId = prop.Value.GetString();
+                                                break;
+                                            }
+                                        }
+                                    }
+                                    break;
+                            }
+                        }
+                        catch (JsonException jex)
+                        {
+#if DEBUG
+                            Debug.WriteLine($"[ConfidentialLedgerFailoverService] JSON parse issue for failoverLedger element: {jex.Message}");
+#endif
+                            _ = jex; // suppress unused warning in non-DEBUG builds
+                        }
+                        catch (InvalidOperationException ioex)
+                        {
+#if DEBUG
+                            Debug.WriteLine($"[ConfidentialLedgerFailoverService] Invalid operation while parsing failoverLedger element: {ioex.Message}");
+#endif
+                            _ = ioex; // suppress unused warning in non-DEBUG builds
+                        }
+
+                        if (!string.IsNullOrEmpty(failoverLedgerId))
+                        {
+                            Uri endpoint = new UriBuilder(primaryEndpoint) { Host = $"{failoverLedgerId}.{LedgerDomainSuffix}" }.Uri;
+                            endpoints.Add(endpoint);
+                        }
+                    }
+                }
+            }
+            catch (Exception)
+            {
+                // ignore entire parse failure
+            }
+            return endpoints;
+        }
+
+        private HttpMessage CreateFailoverRequest(Uri failoverUrl)
+        {
+            HttpMessage message = _pipeline.CreateMessage(new RequestContext(), ResponseClassifier200);
+            Request request = message.Request;
+
+            request.Method = RequestMethod.Get;
+
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(failoverUrl);
+            request.Uri = uri;
+
+            request.Headers.Add("Accept", "application/json");
+
+            return message;
+        }
+
+        private static bool IsRetriableFailure(RequestFailedException ex)
+        {
+            // Include 404 and specific UnknownLedgerEntry error code.
+            return ex.Status == 404 ||
```

Comment:
These errors need to be updated, I doubt that a failover endpoint has so many retriable conditions. 404 would mean that the ledger does not exist, `UnknownLedgerEntry` is not related to `/failover`, 408 not necessary either. 503 and 504 can be dropped because they are included in `ex.Status >= 500`